### PR TITLE
Support public key

### PR DESCRIPTION
**Pull request by asafshen**
## Related Issues

Fixes https://github.com/descope/etc/issues/2981

## Description
Support public key argument to sdk

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -16,9 +16,10 @@ declare const BUILD_VERSION: string;
 /** Configuration arguments which include the Descope core SDK args and an optional management key */
 type NodeSdkArgs = Parameters<typeof createSdk>[0] & {
   managementKey?: string;
+  publicKey?: string;
 };
 
-const nodeSdk = ({ managementKey, ...config }: NodeSdkArgs) => {
+const nodeSdk = ({ managementKey, publicKey, ...config }: NodeSdkArgs) => {
```
Comment (dorsha) [inline lib/index.ts:22]: Can this break compatibility?

File: README.md [hunk 2]
```diff
@@ -860,6 +860,27 @@ of the SDK.
    npm start
    ```
 
+## Providing Custom Public Key
+
+By default, the SDK will download the public key from Descope's servers. You can also provide your own public key. This is useful when the server you are running the SDK on does not have access to the internet.
+
+You can find your public key in the `https://api.descope.com/v2/keys/<project-id>` endpoint.
```
Comment (dorsha) [inline README.md:867]: Maybe we can also add link to keys section in docs.

**Review by dorsha — APPROVED** [review]

File: lib/index.ts (hunk 1 above)
Comment (asafshen) [reply lib/index.ts:22]: no, if you don't provide - it will not do anything

File: README.md (hunk 2 above)
Comment (asafshen) [reply README.md]: good idea

added

**Review by dorsha — APPROVED** [review]

**Pull request merged**